omod-2.5/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_5/OrderAttributeTypeResource2_5.java

```diff
@@ -0,0 +1,103 @@
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Allows standard CRUD for the {@link OrderAttributeType} domain object
+ */
+@Resource(name = RestConstants.VERSION_1 + "/orderattributetype", supportedClass = OrderAttributeType.class, supportedOpenmrsVersions = {
+        "2.5.* - 9.*" })
+public class OrderAttributeTypeResource2_5 extends BaseAttributeTypeCrudResource2_5<OrderAttributeType> {
+	
+	public OrderAttributeTypeResource2_5() {
+	}
+	
+	private OrderService service() {
+		return Context.getOrderService();
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getByUniqueId(java.lang.String)
+	 */
+	@Override
+	public OrderAttributeType getByUniqueId(String uniqueId) {
+		return service().getOrderAttributeTypeByUuid(uniqueId);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#doGetAll(org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	protected NeedsPaging<OrderAttributeType> doGetAll(RequestContext context) throws ResponseException {
+		return new NeedsPaging<OrderAttributeType>(service().getAllOrderAttributeTypes(), context);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#newDelegate()
+	 */
+	@Override
+	public OrderAttributeType newDelegate() {
+		return new OrderAttributeType();
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#save(java.lang.Object)
+	 */
+	@Override
+	public OrderAttributeType save(OrderAttributeType delegate) {
+		return service().saveOrderAttributeType(delegate);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#purge(java.lang.Object,
+	 *      org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	public void purge(OrderAttributeType delegate, RequestContext context) throws ResponseException {
+		service().purgeOrderAttributeType(delegate);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#doSearch(org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	protected NeedsPaging<OrderAttributeType> doSearch(RequestContext context) {
+		List<OrderAttributeType> allAttrs = service().getAllOrderAttributeTypes();
+		List<OrderAttributeType> queryResult = new ArrayList<OrderAttributeType>();
+		for (OrderAttributeType locAttr : allAttrs) {
+			if (Pattern.compile(Pattern.quote(context.getParameter("q")), Pattern.CASE_INSENSITIVE)
```

Comment:
1. We don't really need a regex here. You can just using `locAttr.getName().toLowerCase().contains(context.getParameter("q").toLowerCase())`
2. If we do use a Regex here, do not call `Pattern.compile()` in a loop; lift it to outside the loop context.